ts/src/header-validator/to-json.ts

```diff
@@ -0,0 +1,304 @@
+import * as parsed from './validate-json'
+
+type MaybeHasField<K extends string, V> = {
+  [key in K]?: V
+}
+
+function ifNotNull<K extends string, U, V>(
+  key: K,
+  u: U | null,
+  f: (u: U) => V
+): MaybeHasField<K, V> {
+  const obj: MaybeHasField<K, V> = {}
+  if (u !== null) {
+    obj[key] = f(u)
+  }
+  return obj
+}
+
+export type CommonDebug = {
```

Comment:
Overall LGTM. One comment is that these types are very similar to those defined in validate-json.ts, and it would be nice if they could be reused in some way.